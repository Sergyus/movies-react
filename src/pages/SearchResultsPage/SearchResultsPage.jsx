import React, { useEffect } from "react";
import PropTypes from "prop-types";
import Header from "../../components/Header";
import Menu from "../../components/Menu";
import MoviesContainer from "../../components/Movies";
import styles from "./SearchResultsPage.scss";
import Footer from "../../components/Footer";

const SearchResultsPage = ({ fetchLatestMovies, fetchGenres }) => {
  useEffect(() => {
    fetchGenres();
    fetchLatestMovies(1);
  }, [fetchGenres, fetchLatestMovies]);
  return (
    <>
      <Header />
      <div className={styles.backdrop}>
        <Menu />
        <MoviesContainer />
      </div>
      <Footer />
    </>
  );
};

SearchResultsPage.propTypes = {
  fetchLatestMovies: PropTypes.func,
  fetchGenres: PropTypes.func
};

SearchResultsPage.defaultProps = {
  fetchLatestMovies: () => {},
  fetchGenres: () => {}
};

export default SearchResultsPage;
