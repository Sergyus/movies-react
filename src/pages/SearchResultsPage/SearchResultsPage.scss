.backdrop {
  background-color: #f5f6f6;
  padding-top: 80px;
  padding-bottom: 40px;
}
