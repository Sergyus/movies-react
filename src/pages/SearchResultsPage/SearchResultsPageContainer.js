import { connect } from 'react-redux';
import SearchResultsPage from './SearchResultsPage';
import { fetchGenres, fetchLatestMovies } from '../../modules/app';

export default connect(null, { fetchLatestMovies, fetchGenres })(SearchResultsPage);
