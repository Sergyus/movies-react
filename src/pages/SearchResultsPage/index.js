import SearchResultsPage from './SearchResultsPage';
import SearchResultsPageContainer from './SearchResultsPageContainer';

export { SearchResultsPage, SearchResultsPageContainer };
