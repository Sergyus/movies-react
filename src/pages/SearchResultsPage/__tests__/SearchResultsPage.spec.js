import React from 'react';
import { create, act } from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import SearchResultsPage from '../SearchResultsPage';

describe('SearchResultsPage renders correctly', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const initialState = {
    app: {
      movies: [],
      genres: [],
      isLoading: true,
      currentPage: 1,
      error: false,
      filter: 'Trending',
      currentGenre: '',
      currentSearchQuery: '',
      videoKey: '',
      currentTotalPages: 1,
    },
    ui: {
      layout: 'grid',
    },
  };
  const store = mockStore(initialState);
  it('SearchResultsPage render correctly', () => {
    const tree = create(
      <Provider store={store}>
        <SearchResultsPage />
      </Provider>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('Fetch data calls correctly', () => {
    const mockGenres = jest.fn();
    const mockMovies = jest.fn();
    let tree;
    act(() => {
      tree = create(
        <Provider store={store}>
          <SearchResultsPage fetchGenres={mockGenres} fetchLatestMovies={mockMovies} />
        </Provider>,
      );
    });
    expect(mockGenres).toHaveBeenCalled();
    expect(mockMovies).toHaveBeenCalled();
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
