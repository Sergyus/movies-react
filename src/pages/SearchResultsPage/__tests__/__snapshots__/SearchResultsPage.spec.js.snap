// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsPage renders correctly Fetch data calls correctly 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="container"
    >
      <h1
        className="logo"
      >
        Films
      </h1>
      <form
        className="searchForm"
        onSubmit={[Function]}
      >
        <input
          aria-label="Search Input"
          className="input"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Search..."
          type="search"
          value=""
        />
        <button
          aria-label="Search Button"
          className="searchBtn"
          type="submit"
        />
      </form>
    </div>
  </header>,
  <div
    className="backdrop"
  >
    <section
      className="menu"
      id="menu"
    >
      <nav>
        <ul>
          <li
            className="navItem"
          >
            <button
              aria-label="Trending"
              className="btn btn__active"
              disabled={true}
              onClick={[Function]}
              type="button"
            >
              Trending
            </button>
          </li>
          <li
            className="navItem"
          >
            <button
              aria-label="Top Rated"
              className="btn"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Top Rated
            </button>
          </li>
          <li
            className="navItem"
          >
            <button
              aria-label="Coming Soon"
              className="btn"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Coming Soon
            </button>
          </li>
          <li
            className="navItem"
          >
            <select
              aria-label="Genres"
              className="selectGenre"
              name="selectGenre"
              onChange={[Function]}
              value=""
            >
              <option
                disabled={true}
                hidden={true}
                value=""
              >
                Genre
              </option>
            </select>
          </li>
        </ul>
      </nav>
      <div>
        <button
          aria-label="Grid"
          className="btn btn__active"
          onClick={[Function]}
          type="button"
        />
        <button
          aria-label="Table"
          className="btn"
          onClick={[Function]}
          type="button"
        />
      </div>
    </section>
    <main
      className="container"
    >
      <h2>
        Sorry, no movies found.
      </h2>
    </main>
    <div
      className="spinner"
    >
      <div
        className="ellipsis"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
      <p
        className="title"
      >
        Loading
      </p>
    </div>
  </div>,
  <footer
    className="footer"
  >
    <h2
      className="logo"
    >
      Films
    </h2>
    <p
      className="copyright"
    >
      Copyright © 2019 FILMS. DZMITRY
    </p>
  </footer>,
]
`;

exports[`SearchResultsPage renders correctly SearchResultsPage render correctly 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="container"
    >
      <h1
        className="logo"
      >
        Films
      </h1>
      <form
        className="searchForm"
        onSubmit={[Function]}
      >
        <input
          aria-label="Search Input"
          className="input"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Search..."
          type="search"
          value=""
        />
        <button
          aria-label="Search Button"
          className="searchBtn"
          type="submit"
        />
      </form>
    </div>
  </header>,
  <div
    className="backdrop"
  >
    <section
      className="menu"
      id="menu"
    >
      <nav>
        <ul>
          <li
            className="navItem"
          >
            <button
              aria-label="Trending"
              className="btn btn__active"
              disabled={true}
              onClick={[Function]}
              type="button"
            >
              Trending
            </button>
          </li>
          <li
            className="navItem"
          >
            <button
              aria-label="Top Rated"
              className="btn"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Top Rated
            </button>
          </li>
          <li
            className="navItem"
          >
            <button
              aria-label="Coming Soon"
              className="btn"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Coming Soon
            </button>
          </li>
          <li
            className="navItem"
          >
            <select
              aria-label="Genres"
              className="selectGenre"
              name="selectGenre"
              onChange={[Function]}
              value=""
            >
              <option
                disabled={true}
                hidden={true}
                value=""
              >
                Genre
              </option>
            </select>
          </li>
        </ul>
      </nav>
      <div>
        <button
          aria-label="Grid"
          className="btn btn__active"
          onClick={[Function]}
          type="button"
        />
        <button
          aria-label="Table"
          className="btn"
          onClick={[Function]}
          type="button"
        />
      </div>
    </section>
    <main
      className="container"
    >
      <h2>
        Sorry, no movies found.
      </h2>
    </main>
    <div
      className="spinner"
    >
      <div
        className="ellipsis"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
      <p
        className="title"
      >
        Loading
      </p>
    </div>
  </div>,
  <footer
    className="footer"
  >
    <h2
      className="logo"
    >
      Films
    </h2>
    <p
      className="copyright"
    >
      Copyright © 2019 FILMS. DZMITRY
    </p>
  </footer>,
]
`;
